websites/S/Sonic Retro/presence.ts

```diff
@@ -0,0 +1,279 @@
+const presence = new Presence({
+		clientId: "970743721404530798"
+	}),
+	browsingTimestamp = Math.floor(Date.now());
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+			startTimestamp: browsingTimestamp
+		},
+		{ pathname, search, hostname } = window.location;
+	switch (hostname) {
+		case "sonicretro.org": {
+			presenceData.smallImageKey = "home_512";
+			presenceData.smallImageText = "sonicretro.org";
+			if (search) {
+				presenceData.largeImageKey = "search";
+				if (search === "?s=")
+					presenceData.details = "Searching for something...";
+				else {
+					presenceData.details = `Searching for "${
+						document.title
+							.split("You searched for ")[1]
+							.split(" - Sonic Retro")[0]
+					}"`;
+				}
+			} else if (
+				(pathname === ("/" || "/#") && !search) ||
+				(pathname.includes("/page/") && !search)
+			) {
+				presenceData.largeImageKey = "home";
+				presenceData.details = "Zooming through the main page";
+			} else if (pathname.includes("/category/")) {
+				presenceData.largeImageKey = "search";
+				presenceData.details = `Looking through ${document.title
+					.split("-")[0]
+					.toLowerCase()}`;
+			} else if (
+				document.title.match(
+					/January|February|March|April|May|June|July|August|September|October|November|December/g
+				)
+			) {
+				presenceData.largeImageKey = "search";
+				presenceData.details = `Looking for posts in/on ${
```

Comment:
why in/on?

Reply:
Can I see an example of this with the variable replaced?

Reply:
> Can I see an example of this with the variable replaced?

![image](https://user-images.githubusercontent.com/96930330/169417998-ea0df37c-d7af-4cc1-a6fa-cb6cead8e8a3.png)
![image](https://user-images.githubusercontent.com/96930330/169418021-116ba3e5-78c6-4752-95ce-7fe9c276f266.png)

Reply:
```suggestion
				presenceData.details = `Looking for posts from ${
```